src/main/java/io/cryostat/core/reports/InterruptibleReportGenerator.java

```diff
@@ -185,6 +164,68 @@ public Future<ReportResult> generateReportInterruptibly(
                 });
     }
 
+    public Future<Map<String, RuleEvaluation>> generateEvalMapInterruptibly(
+            InputStream recording, Predicate<IRule> predicate) {
+        Objects.requireNonNull(recording);
+        Objects.requireNonNull(predicate);
+        return qThread.submit(
+                () -> {
+                    try {
+                        Collection<Result> results =
+                                generateResultHelper(recording, predicate).left;
+                        Map<String, RuleEvaluation> evalMap = new HashMap<String, RuleEvaluation>();
+                        for (var eval : results) {
+                            evalMap.put(
+                                    eval.getRule().getId(),
+                                    new RuleEvaluation(
+                                            eval.getScore(),
+                                            eval.getRule().getName(),
+                                            eval.getRule().getTopic(),
+                                            eval.getShortDescription()));
+                        }
+                        return evalMap;
+                    } catch (InterruptedException
+                            | IOException
+                            | ExecutionException
+                            | CouldNotLoadRecordingException e) {
+                        return Map.of(
+                                e.getClass().toString(),
+                                new RuleEvaluation(e.hashCode(), "", "", e.getMessage()));
```

Comment:
The `hashCode` here isn't particularly meaningful - what benefit does that give to a client to receive that information? Perhaps that should be `NaN` or `infinity` or something instead.

Probably also worth logging the exception here too.